.github/workflows/publish.yml

```diff
@@ -1,63 +1,110 @@
-          echo "::set-output name=isrelease::${{ steps.split-qualifier.outputs.replaced == 'VERSIONRELEASE' || steps.split-qualifier.outputs.replaced == 'VERSION'  }}"
-          echo "::set-output name=isunknown::${{ startsWith(steps.split-qualifier.outputs.replaced,'VERSION') != true  }}"
-    - name: validate version interpretation
-      if: steps.version-type.outputs.isunknown == 'true'
+      id: version
+      #we only run the qualifyVersionGha task so that no other console printing can hijack this step's output
+      #output: versionType, fullVersion
+      run: ./gradlew qualifyVersionGha
+    - name: fail-if-bad
+      if: steps.version.outputs.versionType == 'BAD'
       run: |
-          echo "Couldn't classify version"
-          echo "${{ steps.read-version.outputs.value }}"
-          echo "${{ toJson(steps.split-qualifier.outputs) }}"
-          echo "${{ toJson(steps.version-type.outputs) }}"
+          echo "Couldn't classify version ${{ steps.version.outputs.fullVersion }}, see warning in above step"
           return 1
+    - name: run checks
+      id: checks
+      run: ./gradlew check
+    - name: tag
+      uses: avakar/tag-and-release@v1
+      if: steps.version.outputs.versionType == 'RELEASE' || steps.gradle.outputs.versionType == 'MILESTONE'
+      with:
+        tag_name: "v${{ steps.version.outputs.fullVersion }}"
+        draft: true
+        prerelease: ${{ steps.version.outputs.versionType == 'MILESTONE' }}
+        body: "Reactor Addons `${{ steps.version.outputs.fullVersion }}` is part of the **`RELEASETRAIN` release train (codename `CODENAME`)**. TODO"
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  #deploy the snapshot artifacts to Artifactory
+  deploySnapshot:
+    name: deploySnapshot
+    runs-on: ubuntu-latest
+    needs: prepare
+    if: needs.prepare.outputs.versionType == 'SNAPSHOT'
+    environment: snapshots
+    steps:
+    - uses: actions/checkout@v2
     - uses: actions/setup-java@v1
       with:
         java-version: 8
-    - uses: eskatos/gradle-command-action@v1
-      name: gradle
+    - name: deploy
       env:
-        ORG_GRADLE_PROJECT_artifactory_publish_username: ${{secrets.ORG_GRADLE_PROJECT_ARTIFACTORY_PUBLISH_USERNAME}}
-        ORG_GRADLE_PROJECT_artifactory_publish_password: ${{secrets.ORG_GRADLE_PROJECT_ARTIFACTORY_PUBLISH_PASSWORD}}
+        ORG_GRADLE_PROJECT_artifactory_publish_username: ${{secrets.ARTIFACTORY_SNAPSHOT_USERNAME}}
+        ORG_GRADLE_PROJECT_artifactory_publish_password: ${{secrets.ARTIFACTORY_PASSWORD}}
+      run: |
+          ./gradlew assemble artifactoryPublish -Partifactory_publish_contextUrl=https://repo.spring.io -Partifactory_publish_repoKey=libs-snapshot-local
+
+  #sign the milestone artifacts and deploy them to Artifactory
+  deployMilestone:
+    name: deployMilestone
+    runs-on: ubuntu-latest
```

Comment:
Would not recommend `-latest`. Imagine fighting recently changed GHA environment during a release routine

Reply:
updated